provisioner/ansible/provisioner.go

```diff
@@ -327,12 +328,21 @@ func (p *Provisioner) executeAnsible(ui packer.Ui, comm packer.Communicator, pri
 
 	wg := sync.WaitGroup{}
 	repeat := func(r io.ReadCloser) {
-		scanner := bufio.NewScanner(r)
-		for scanner.Scan() {
-			ui.Message(scanner.Text())
-		}
-		if err := scanner.Err(); err != nil {
-			ui.Error(err.Error())
+		reader := bufio.NewReader(r)
+		for {
+			line, err := reader.ReadString('\n')
+			if line != "" {
+				line = strings.TrimRightFunc(line, unicode.IsSpace)
+				ui.Message(line)
+			}
+			if err != nil {
+				if err == io.EOF {
+					break
+				} else {
```

Comment:
Maybe remove this useless else and move the err block before checking the line at all?